Assets/Scripts/Enemy.cs

```diff
@@ -12,8 +12,13 @@ public class Enemy : MonoBehaviour, IDamageable
     public float speed;
 
     [Tooltip("Current waypoint this enemy should move towards")]
-    public Transform targetWaypoint;
+    private Transform targetWaypoint;
+    private int waypointIndex;
+    public GameObject waypoints;
```

Comment:
You can reference the Waypoint script directly to reduce the number of GetComponent calls.